src/de.rs

```diff
@@ -450,18 +450,21 @@ impl<'de, 'a, R: dec::Read<'de>> Accessor<'a, R> {
     #[inline]
     pub fn tuple(
         de: &'a mut Deserializer<R>,
-        len: usize,
+        exp_len: usize,
     ) -> Result<Accessor<'a, R>, DecodeError<R::Error>> {
         let array_start = dec::ArrayStart::decode(&mut de.reader)?;
-
-        if array_start.0 == Some(len) {
-            Ok(Accessor { de, len })
+        if let Some(len) = array_start.0 {
+            if len <= exp_len {
+                Ok(Accessor { de, len })
+            } else {
```

Comment:
IMO, it's ok to simply not check this at all. The caller is expected read each element off the `Accessor` and handle cases where it's too short.